diagnostics/block.go

```diff
@@ -0,0 +1,116 @@
+package diagnostics
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+	"strconv"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+
+	"github.com/treeverse/lakefs/config"
+
+	"github.com/treeverse/lakefs/block"
+
+	"github.com/treeverse/lakefs/db"
+
+	"github.com/treeverse/lakefs/catalog"
+)
+
+// DBCollector collects diagnostics information and write the collected content into a writer in a zip format
+type BlockCollector struct {
+	cataloger   catalog.Cataloger
+	adapter     block.Adapter
+	cfg         *config.Config
+	dbCollector *DBCollector
+}
+
+// NewDBCollector accepts database to work with during collect
+func NewBlockCollector(adb db.Database, cataloger catalog.Cataloger, cfg *config.Config, adapter block.Adapter) *BlockCollector {
+	return &BlockCollector{
+		cataloger:   cataloger,
+		cfg:         cfg,
+		adapter:     adapter,
+		dbCollector: NewDBCollector(adb),
```

Comment:
why one collector holds another?

Reply:
So that the `DBCollector` be a standalone unit like today, and the `BlockCollector` wraps it and adds graveler statistics on top of it.